src/android/CameraLauncher.java

```diff
@@ -173,6 +181,20 @@ else if ((srcType == PHOTOLIBRARY) || (srcType == SAVEDPHOTOALBUM)) {
     // LOCAL METHODS
     //--------------------------------------------------------------------------
 
+    private void processConfiguration(JSONArray args) throws JSONException {
+        if (args == null)
+            throw new JSONException("no configuration object passed");
+        this.mQuality = args.optInt(0, 80);
+        // args 1 and 2 (destType, srcType) are skipped because they have no field representation in this class
+        this.targetWidth = args.optInt(3);
+        this.targetHeight = args.optInt(4);
+        this.encodingType = args.optInt(5, JPEG);
+        this.mediaType = args.optInt(6, PICTURE);
+        this.allowEdit = args.optBoolean(7);
+        this.correctOrientation = args.optBoolean(8);
+        this.saveToPhotoAlbum = args.optBoolean(9);
+    }
```

Comment:
Nice! FWIW, I added in defaults for all values. Here's my edit:

        this.mQuality = args.optInt(0, 80);
        // args 1 and 2 (destType, srcType) are skipped because they have no field representation in this class
        this.targetWidth = args.optInt(3, 0);
        this.targetHeight = args.optInt(4, 0);
        this.encodingType = args.optInt(5, JPEG);
        this.mediaType = args.optInt(6, PICTURE);
        this.allowEdit = args.optBoolean(7, false);
        this.correctOrientation = args.optBoolean(8, false);
        this.saveToPhotoAlbum = args.optBoolean(9, false);